solang-parser/src/test.rs

```diff
@@ -1003,3 +1007,113 @@ contract C {
     let (actual_parse_tree, _) = crate::parse(src, 0).unwrap();
     assert_eq!(actual_parse_tree.0.len(), 1);
 }
+
+#[test]
+fn test_libsolidity() {
+    fn timeout_after<T, F>(d: Duration, f: F) -> Result<T, String>
+    where
+        T: Send + 'static,
+        F: FnOnce() -> T,
+        F: Send + 'static,
+    {
+        let (done_tx, done_rx) = mpsc::channel();
+        let handle = thread::spawn(move || {
+            let val = f();
+            done_tx.send(()).expect("Unable to send completion signal");
+            val
+        });
+
+        match done_rx.recv_timeout(d) {
+            Ok(_) => Ok(handle.join().expect("Thread panicked")),
+            Err(_) => Err(format!("Thread timeout-ed after {d:?}")),
+        }
+    }
+
+    let source_delimiter = regex::Regex::new(r"====.*====").unwrap();
+    let error_matcher = regex::Regex::new(r"// ----\n// \w+( \d+)?:").unwrap();
```

Comment:
On Windows, this regex fails because the files are checked out with dos end of lines (`\r\n`).

The regex should it should be: `r"// ----\r?\n// \w+( \d+)?:"` and then the tests pass (on my windows machine at least)

Reply:
fixed